x/ibc-hooks/contracts/crosschain-swaps/src/msg.rs

```diff
@@ -0,0 +1,89 @@
+use cosmwasm_schema::{cw_serde, QueryResponses};
+use cosmwasm_std::{Addr, Coin};
+use swaprouter::msg::Slippage;
+
+/// Message type for `instantiate` entry_point
+#[cw_serde]
+pub struct InstantiateMsg {
+    /// This should be an instance of the Osmosis swaprouter contract
+    pub swap_contract: String,
+    /// If set to true, the contract will add a callback request on the packet
+    /// so that it gets notified when an ack is received or if the packet timed
+    /// out. If set to false, any funds sent on a packet that fails after a swap
+    /// will be stuck in this contract.
+    ///
+    /// The information about the packet sender and recovery address is still
+    /// stored, so recovery could be possible after a contract upgrade.
+    pub track_ibc_sends: Option<bool>,
+    /// These are the channels that will be accepted by the contract. This is
+    /// needed to avoid sending packets to addresses not supported by the
+    /// receiving chain. The channels are specified as (bech32_prefix, channel_id)
+    pub channels: Vec<(String, String)>,
+}
+
+#[cw_serde]
+pub struct Recovery {
+    /// An osmosis addres used to recover any tokens that get stuck in the
+    /// contract due to IBC failures
+    pub recovery_addr: Addr,
+}
+
+/// Message type for `execute` entry_point
+#[cw_serde]
+pub enum ExecuteMsg {
+    /// Execute a swap and forward it to the receiver address on the specified ibc channel
+    OsmosisSwap {
+        input_coin: Coin,
+        output_denom: String,
+        receiver: Addr,
+        slippage: Slippage,
+        next_memo: Option<String>,
```

Comment:
what is next_memo?

Reply:
that's the memo to use in the forwarded packet. I'll document that.